modules/proxy-certificates.adoc

```diff
@@ -0,0 +1,114 @@
+
+// Module included in the following assemblies:
+//
+// * authentication/understanding-certificates.adoc
+
+[id="proxy-certificates_{context}"]
+= Proxy certificates
+
+== Purpose
+
+The `trustedCA` field of `proxy` is a reference to a ConfigMap that contains a
+user-provided trusted certificate authority (CA) bundle. This bundle is merged
+with the Red Hat Enterprise Linux CoreOS (RHCOS) trust bundle and injected into
+the trust store of platform components that make egress HTTPS calls. For
+example, `image-registry-operator` calls an external image registry to download
+images. If `trustedCA` is not specified, only the RHCOS trust bundle is used for
+proxied HTTPS connections. Provide custom CA certs to the RHCOS trust bundle if
+you want to use your own certificate infrastructure.
+
+The `trustedCA` field should only be consumed by a proxy validator. The
+validator is responsible for reading the certificate bundle from required key
+`ca-bundle.crt` and copying it to a ConfigMap named `trusted-ca-bundle` in the
+`openshift-config-managed` namespace. The namespace for the ConfigMap referenced
+by `trustedCA` is `openshift-config`:
+
+[source,yaml]
+----
+apiVersion: v1
+	kind: ConfigMap
+ metadata:
+	 name: user-ca-bundle
+	 namespace: openshift-config
+	 data:
+	   ca-bundle.crt: |
+	     -----BEGIN CERTIFICATE-----
+	     Custom CA certificate bundle.
+	     -----END CERTIFICATE-----
+----
+
+== File path
+
+The user-provided trust bundle is represented as a ConfigMap. The ConfigMap is
+mounted into the filesystem of platform components that make egress HTTPS calls.
+Typically, Operators mount the ConfigMap to
+`/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`, but this not required by
+proxy and is at the Operator's discretion.
+
+If using the RHCOS trust bundle, place CA certificates in
+`/etc/pki/ca-trust/source/anchors`.
```

Comment:
It might be good to point users at the RHEL 8 documentation for what were are doing. 

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/using-shared-system-certificates_security-hardening

This not only tells them that were following the RHEL documented way of updating the host's/containers(not sure which) trust bundle; but we are following a defiend/documented best practice.

Reply:
I find the following confusing:
```
If using the RHCOS trust bundle, place CA certificates in
`/etc/pki/ca-trust/source/anchors`.
```

We tell suers to use the proxy object to add custom ca certs and then contradict ourselves by the above statement. I think it's good to have an xref to the  rhcos cert docs after L67.

Reply:
@ahardin-rh I see the PR still has the following:

```
If using the RHCOS trust bundle, place CA certificates in
`/etc/pki/ca-trust/source/anchors`.
```

As I mention above, I find this to be confusing.